### PR TITLE
Yasmin-Leaves

### PR DESCRIPTION
**Pull request by YasminM11**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The Model interact with the database and create all the business logic for the application.
Describe in your own words what the Controller is doing in Rails | the controller is the logic center that handle the interaction between other parts of the server.
Describe in your own words what the View is doing in Rails | view is the window to the user to see all the logic and code, contains presentation logic to generate HTML.
Describe an edge-case controller test you wrote |An edge-case controller test I wrote was a nonexistent task.
What is the purpose of using strong params? (i.e. the params method in the controller) | using strong params helps to Dry the code, it use in our controllers to prevent errors and to provide a safe way to require and permit data that comes from our forms.
How are Rails migrations related to Rails models? | Rails migrations give the way to modify database schema.
Describe one area of Rails that are still unclear on | I am still not comfortable to handle all the files and make them interacting with each other.

### CODE REVIEW

File: app/views/tasks/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <% @tasks.each do |task| %>
+    <li>
+      <strong>Task:</strong> <%= link_to "#{task.name}", task_path(task)%>
+      <strong>Description:</strong> <%="#{task.description}"%>
+      <strong>Date:</strong> <%= "#{task.date}" %> 
+
+      <div><br>
+      <%= link_to  "Edit Task | ", edit_task_path(task) %>
+      <%= link_to  "Delete Task | ", task_path(task), method: :delete, data: {confirm: "Are you sure you want to delete?"}%><br>
+      <%= link_to "Task completed | ", completed_path(task), method: :patch %>
```
Comment (beccaelenzil) [inline app/views/tasks/index.html.erb:16]: There is a small bug here. Both the "Task completed" and the "Task Not Completed" link toggle the completed setting to the opposite setting that it was currently. These buttons need to have a reliable behavior. One way to achieve this, would be to have 2 distinct controller actions: one `mark_incomplete` the other `mark_complete`.

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -3,124 +3,197 @@
...
-
-  end
-
-  # Complete for Wave 4
-  describe "toggle_complete" do
+  
+   # Complete these tests for Wave 4
+    describe "destroy" do
     # Your tests go here
+    it "can delete an existing task" do
```
Comment (beccaelenzil) [inline test/controllers/tasks_controller_test.rb:163]: This test looks great. It is important to also test the edge case of trying to destroy a task with an invalid id.

File: test/controllers/tasks_controller_test.rb [hunk 3]
```diff
@@ -3,124 +3,197 @@
...
+      must_redirect_to root_path
+      
+      #Assert
+      expect( Task.find_by(id: existing_task.id) ).must_be_nil
+    end
+  
+ end
+  
+  # #   # Complete for Wave 4
+   describe "toggle_complete" do
```
Comment (beccaelenzil) [inline test/controllers/tasks_controller_test.rb:178]: It is also important to test the edge case of trying to delete a task with an invalid id.

**Comment by beccaelenzil** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | good work on this!
Answered comprehension questions | check
Successfully handles: Index, Show | check
Index & Show tests pass |check
Successfully handles: New, Create | check
New & Create tests pass |check
Successfully handles: Edit, Update |check
Tests for Edit & Update test valid & invalid task ids | check
Successfully handles: Destroy, Task Complete | see comment about the task complete functionality -- there is a small bug
Tests for Destroy & Task Complete include tests for valid and invalid task ids | see comment about testing for invalid task ids
Routes follow RESTful conventions | check
Uses named routes (like `_path`) | check
**Overall** | Good work overall. You hit the learning goals of using partials, using strong params, and following Rails conventions. There is a bug in the logic for the toggle_complete implementation that makes it so that when you press `mark complete` it will mark the task incomplete if it was already complete. Additionally, you need to make sure you test valid and edge cases -- in this case this means testing all methods with invalid ids.
